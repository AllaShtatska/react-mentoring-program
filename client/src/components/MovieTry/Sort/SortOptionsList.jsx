import '../../../static/inline';
import '../../../static/margin-left';

import React from 'react';
import SortOption from './SortOption';
import sortType from './sortType';


class SortOptionsList extends React.Component{
    render(){
        return(
            <div className="inline margin-left">
                Sort By
                <SortOption currentSortType={this.props.sortType} sortType={sortType.releaseDate} onSwitchSortType={this.props.onSwitchSortType}/>
                <SortOption currentSortType={this.props.sortType} sortType={sortType.rating} onSwitchSortType={this.props.onSwitchSortType}/>
            </div>
        );
    }
}

export default SortOptionsList;