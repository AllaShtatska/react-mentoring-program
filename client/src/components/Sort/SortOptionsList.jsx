import '../../static/inline.css';
import '../../static/margin-left.css';

import React from 'react';
import PropTypes from 'prop-types';
import SortOption from './SortOption';
import SORT_TYPE from './sortType';


const SortOptionsList = ({ currentSortType, onSwitchSortType }) => {
    const sortOptions = Object.values(SORT_TYPE).map(sortType => (
      <SortOption
        key={sortType}
        currentSortType={currentSortType}
        sortType={sortType}
        onSwitchSortType={onSwitchSortType}
     />
    ));
    
    return(
      <div className="inline margin-left">
        Sort By
        { sortOptions }
      </div>
    );
  };

  SortOptionsList.propTypes = {
    currentSortType: PropTypes.string.isRequired,
    onSwitchSortType: PropTypes.func.isRequired,
  };

  export default SortOptionsList;