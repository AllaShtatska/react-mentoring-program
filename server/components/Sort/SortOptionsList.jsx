import inlineStyles from "../../static/inline.css";
import marginLeftStyles from "../../static/margin-left.css";

import React from "react";
import PropTypes from "prop-types";
import SortOption from "./SortOption";
import SORT_TYPE from "./sortType";

const SortOptionsList = ({ currentSortType, onSwitchSortType }) => {
  const sortOptions = Object.values(SORT_TYPE).map(sortType => (
    <SortOption
      key={sortType}
      currentSortType={currentSortType}
      sortType={sortType}
      onSwitchSortType={onSwitchSortType}
    />
  ));

  const className = inlineStyles.inline + " " + marginLeftStyles["margin-left"];

  return (
    <div className={className}>
      Sort By
      {sortOptions}
    </div>
  );
};

SortOptionsList.propTypes = {
  currentSortType: PropTypes.string.isRequired,
  onSwitchSortType: PropTypes.func.isRequired
};

export default SortOptionsList;
