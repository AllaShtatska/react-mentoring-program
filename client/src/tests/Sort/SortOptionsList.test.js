import React from "react";
import renderer from "react-test-renderer";
import SortOptionsList from "../../components/Sort/SortOptionsList";
import SORT_TYPE from "../../components/Sort/sortType";

it("renders correctly", () => {
  const component = renderer.create(<SortOptionsList currentSortType={SORT_TYPE.rating} onSwitchSortType={function () {}}/>);

  const jsonComponent = component.toJSON();
  expect(jsonComponent).toMatchSnapshot();
});
