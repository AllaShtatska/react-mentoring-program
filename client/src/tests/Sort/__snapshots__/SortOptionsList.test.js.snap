// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="inline margin-left"
>
  Sort By
  <div
    className="inline"
  >
    <button
      className="sort-button-deselected"
      onClick={[Function]}
    >
      release date
    </button>
  </div>
  <div
    className="inline"
  >
    <button
      className="sort-button-selected"
      onClick={[Function]}
    >
      rating
    </button>
  </div>
</div>
`;
